bin/perf.c

```diff
@@ -0,0 +1,140 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+#include <s2n.h>
+#include <stdlib.h>
+#include <string.h>
+#include <stdio.h>
+
+#include "error/s2n_errno.h"
+#include "stuffer/s2n_stuffer.h"
+#include "utils/s2n_endian.h"
+#include "utils/s2n_safety.h"
+#include "common.h"
+
+uint64_t read_uint64(uint8_t *data)
+{
+    uint64_t u = 0;
+    u |= ((uint64_t) data[0]);
+    u |= ((uint64_t) data[1]) << (1 * 8);
+    u |= ((uint64_t) data[2]) << (2 * 8);
+    u |= ((uint64_t) data[3]) << (3 * 8);
+    u |= ((uint64_t) data[4]) << (4 * 8);
+    u |= ((uint64_t) data[5]) << (5 * 8);
+    u |= ((uint64_t) data[6]) << (6 * 8);
+    u |= ((uint64_t) data[7]) << (7 * 8);
+    u = be64toh(u);
+    return u;
+}
+
+void write_uint64(uint8_t *data, uint64_t u)
+{
+    u = htobe64(u);
```

Comment:
Is this necessary? You couldn't build S2N on a big-endian system rn; we assume little endian and always translate to network order before sending over the wire